src/content/docs/en/reference/api-reference.mdx

```diff
@@ -537,6 +537,18 @@ import Shout from "../components/Shout.astro";
 <!-- renders as <div>SLOTS!</div> -->
 ```
 
+Callback functions can be passed to named slots using a HTML tag with a `slot` attribute:
+
+```astro
+<Shout message="slots!">
+  <fragment slot="message">
+    {(message) => <div>{message}</div>}
+  </fragment>
+</Shout>
+```
+
+This tag cannot be named `Fragment`, `slot` or include a dash `-`. The tag is only used to transfer the callback to a named slot, it does not get rendered onto the page.
```

Comment:
```suggestion
Use a standard HTML element for the wrapping tag and not a component such as `<Fragment />`.  However, you cannot use the HTML `<slot>` element as this will be interpreted as an Astro slot. Lastly, the element name may not include a dash `-` to avoid conflicts with data attributes.
```

I'm wondering whether we can get away with just saying "Use a standard HTML element" which solves most of the naming restrictions. :smile:  Like, if they do choose a standard HTML element (that's not slot), we don't have to worry about them using Fragment (because it's a component, not an HTML element). I added a reason (not sure if accurate?) about why no dashes, but maybe we don't even need that sentence *if* we assume that a standard HTML element doesn't have dashes in the first place?

Reply:
Ya, I like that, that makes sense. I took another look at the dash `-` restriction and it looks like a bug that can be fixed upstream, so we can just remove that restriction to simplify things:

```suggestion
Always use a standard HTML element for the wrapping tag and not a `<Fragment />` or component. However, you cannot use the HTML element `<slot>` as this will be interpreted as an Astro slot. 
```